sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/NettyAsyncHttpClient.java

```diff
@@ -153,11 +158,24 @@
                     hdr.getValuesList().forEach(value -> reactorNettyRequest.addHeader(hdr.getName(), value));
                 }
             }
-            if (restRequest.getBody() != null) {
+
+            if (restRequest.getContent() == null) {
+                return reactorNettyOutbound;
+            }
+
+            BinaryDataContent binaryDataContent = BinaryDataHelper.getContent(restRequest.getContent());
+            if (binaryDataContent instanceof ByteArrayContent) {
+                return reactorNettyOutbound.sendByteArray(Mono.defer(() -> Mono.just(binaryDataContent.toBytes())));
```

Comment:
does defer make sense here? the content would have the byte array already in memory at this point I think.